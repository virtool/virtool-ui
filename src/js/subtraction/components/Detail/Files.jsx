import { map, sortBy } from "lodash-es";
import React from "react";
import { BoxGroup, BoxGroupHeader, NoneFound } from "../../../base";
import { File as SubtractionFile } from "./File";

export default function SubtractionFiles({ files }) {
    let fileComponents = map(sortBy(files, "name"), file => <SubtractionFile file={file} key={file.id} />);

    if (files.length === 0) {
        fileComponents = <NoneFound noun="subtraction files" />;
    }

    return (
        <BoxGroup>
            <BoxGroupHeader>
                <h2>Files</h2>
                <p>Data files available to workflows using this subtraction.</p>
            </BoxGroupHeader>
            {fileComponents}
        </BoxGroup>
    );
}
