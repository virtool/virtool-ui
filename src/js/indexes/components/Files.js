import { map } from "lodash-es";
import React from "react";
import { connect } from "react-redux";
import { Badge, BoxGroup, BoxGroupHeader } from "../../base";
import { File as IndexFile } from "../../subtraction/components/Detail/File";

export const Files = ({ files }) => {
    const fileComponents = map(files, file => <IndexFile file={file} key={file.id} />);
    return (
        <BoxGroup>
            <BoxGroupHeader>
                <h2>
                    Files
                    <Badge>{files.length}</Badge>
                </h2>
                Data files available to workflows using this index.
            </BoxGroupHeader>
            {fileComponents}
        </BoxGroup>
    );
};

const mapStateToProps = state => ({
    files: state.indexes.detail.files
});

export default connect(mapStateToProps)(Files);
