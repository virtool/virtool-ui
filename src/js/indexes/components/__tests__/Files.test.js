import { Files } from "../Files";
import { screen } from "@testing-library/react";

describe("<IndexDetail />", () => {
    let props;

    beforeEach(() => {
        props = {
            files: [
                { id: "1", name: "foo", download_url: "/testUrl/foo", size: "1024" },

                { id: "2", name: "bar", download_url: "/testUrl/bar", size: "2048" }
            ]
        };
    });

    it("should render", () => {
        const wrapper = shallow(<Files {...props} />);
        expect(wrapper).toMatchSnapshot();
    });

    it("should show component", () => {
        renderWithProviders(<Files {...props} />);
        expect(screen.getByText("Files")).toBeInTheDocument();
    });

    it("should render names", () => {
        renderWithProviders(<Files {...props} />);
        expect(screen.getByText("foo")).toBeInTheDocument();
    });

    it("should render sizes", () => {
        renderWithProviders(<Files {...props} />);
        expect(screen.getByText("1.0KB")).toBeInTheDocument();
    });
    it("should include url", () => {
        renderWithProviders(<Files {...props} />);
        expect(screen.getAllByRole("link", {})[0]).toHaveAttribute("href", "/api/testUrl/foo");
    });
});
