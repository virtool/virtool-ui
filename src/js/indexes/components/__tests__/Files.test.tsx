import { screen } from "@testing-library/react";
import React from "react";
import { describe, expect, it } from "vitest";
import { renderWithProviders } from "../../../../tests/setupTests";
import { Files } from "../Files";

describe("<Files />", () => {
    it("should render", () => {
        const files = [
            {
                id: "1",
                name: "foo",
                download_url: "/testUrl/foo",
                size: "1024",
            },
            {
                id: "2",
                name: "bar",
                download_url: "/testUrl/bar",
                size: "2048",
            },
        ];

        renderWithProviders(<Files files={files} />);

        const heading = screen.getByRole("heading", {
            level: 2,
        });

        expect(heading).toHaveTextContent("Files");
        expect(heading).toHaveTextContent("2");

        const foo = screen.getByText("foo");

        expect(foo.closest("div")).toHaveTextContent("1.0KB");
        expect(foo.closest("div")).toHaveTextContent("foo");
        expect(foo.closest("a")).toHaveAttribute("href", "/api/testUrl/foo");

        const bar = screen.getByText("bar");

        expect(bar.closest("div")).toHaveTextContent("2.0KB");
        expect(bar.closest("div")).toHaveTextContent("bar");
        expect(bar.closest("a")).toHaveAttribute("href", "/api/testUrl/bar");
    });
});
