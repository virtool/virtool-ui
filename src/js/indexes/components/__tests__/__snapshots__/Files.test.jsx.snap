// Vitest Snapshot v1

exports[`<IndexDetail /> > should render 1`] = `
<BoxGroup>
  <BoxGroupHeader>
    <h2>
      Files
      <Badge>
        2
      </Badge>
    </h2>
    Data files available to workflows using this index.
  </BoxGroupHeader>
  <File
    file={
      {
        "download_url": "/testUrl/foo",
        "id": "1",
        "name": "foo",
        "size": "1024",
      }
    }
    key="1"
  />
  <File
    file={
      {
        "download_url": "/testUrl/bar",
        "id": "2",
        "name": "bar",
        "size": "2048",
      }
    }
    key="2"
  />
</BoxGroup>
`;
